### PR TITLE
#368 [Grid] Set Configuration as favorite

### LINKED ISSUE #368: [Grid] Set grid configuration as favorite

### PR DESCRIPTION
**Pull request by martineiber**
## Changes in this pull request
Resolves #368

### CODE REVIEW

**Comment by martineiber** [comment]
@lukmzig @mattamon Any Idea how to solve this nicely? Or Accept it?
https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=386&id=pimcore_studio-backend-bundle&open=AZGODXB5Q4zOCu9FD1zb

**Comment by mattamon** [comment]
> @lukmzig @mattamon Any Idea how to solve this nicely? Or Accept it? https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=386&id=pimcore_studio-backend-bundle&open=AZGODXB5Q4zOCu9FD1zb

Not really. I mean you could try only to check for the false and return false one time and the other times true.

You could join all the coniditions with an OR. 

You could also for a match, but I do not see how this would make it easier. :/

File: src/Grid/Service/FavoriteService.php [hunk 1]
```diff
@@ -68,4 +79,51 @@ public function removeAssetConfigurationAsFavoriteForCurrentUser(
 
         return $gridConfiguration;
     }
+
+    private function isCurrentUserAllowsToSetAsFavorite(GridConfiguration $gridConfiguration): bool
+    {
+        if ($gridConfiguration->getOwner() === $this->securityService->getCurrentUser()->getId()) {
```
Comment (lukmzig) [inline src/Grid/Service/FavoriteService.php]: Would it not be better to get the current user once and then pass him to private methods? WDYT?

Comment (martineiber) [reply src/Grid/Service/FavoriteService.php]: Good Point! I will change this :)

**Comment by martineiber** [comment]
> > @lukmzig @mattamon Any Idea how to solve this nicely? Or Accept it? https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=386&id=pimcore_studio-backend-bundle&open=AZGODXB5Q4zOCu9FD1zb
> 
> Not really. I mean you could try only to check for the false and return false one time and the other times true.
> 
> You could join all the coniditions with an OR.
> 
> You could also for a match, but I do not see how this would make it easier. :/

That would negatively affect the readability everywhere IMHO.

**Comment by lukmzig** [comment]
> > > @lukmzig @mattamon Any Idea how to solve this nicely? Or Accept it? https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=386&id=pimcore_studio-backend-bundle&open=AZGODXB5Q4zOCu9FD1zb
> > 
> > 
> > Not really. I mean you could try only to check for the false and return false one time and the other times true.
> > You could join all the coniditions with an OR.
> > You could also for a match, but I do not see how this would make it easier. :/
> 
> That would negatively affect the readability everywhere IMHO.

I would also say so ...we should just accept it. Here its pretty clear

**Comment by mattamon** [comment]
> > > @lukmzig @mattamon Any Idea how to solve this nicely? Or Accept it? https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=386&id=pimcore_studio-backend-bundle&open=AZGODXB5Q4zOCu9FD1zb
> > 
> > 
> > Not really. I mean you could try only to check for the false and return false one time and the other times true.
> > You could join all the coniditions with an OR.
> > You could also for a match, but I do not see how this would make it easier. :/
> 
> That would negatively affect the readability everywhere IMHO.

Yeah, also exactly my thought tbh. Tha's why I said not really. 

Let's accept it, we have other places like this too where we accepted it as is.

File: src/Grid/Service/FavoriteService.php (hunk 1 above)
Comment (martineiber) [reply src/Grid/Service/FavoriteService.php]: Fixed here https://github.com/pimcore/studio-backend-bundle/pull/386/commits/70d12f4a443291c090f59c26b4a06c76097383ba

**Review by lukmzig — APPROVED** [review]
favorite PR of the 🐧 :star:

**Pull request merged**